homeassistant/components/mqtt/debug_info.py

```diff
@@ -87,8 +87,10 @@ def add_entity_discovery_data(hass, discovery_data, entity_id):
 
 def update_entity_discovery_data(hass, discovery_payload, entity_id):
     """Update discovery data."""
-    entity_info = hass.data[DATA_MQTT_DEBUG_INFO]["entities"][entity_id]
-    entity_info["discovery_data"][ATTR_DISCOVERY_PAYLOAD] = discovery_payload
+    entity_info = hass.data[DATA_MQTT_DEBUG_INFO]["entities"].get(entity_id)
+    if entity_info is not None:
+        entity_info["discovery_data"][ATTR_DISCOVERY_PAYLOAD] = discovery_payload
+    return entity_info
```

Comment:
What changed to make us need the guard here?

Reply:
The test_cleanup_device in test_discovery.py will fail. This is caused by RestoreEntity, that already cleared the registry. I have tested with a switch entity, and that will give the same behavior. So this is the work a round for the bug.